packages/next-auth/src/core/types.ts

```diff
@@ -325,7 +326,7 @@ export interface CallbacksOptions<P = Profile, A = Account> {
     session: Session
     user: User | AdapterUser
     token: JWT
-  }) => Awaitable<Session>
+  }) => Awaitable<Session|Falsy>
   /**
    * This callback is called whenever a JSON Web Token is created (i.e. at sign in)
```

Comment:
The comment above needs to be extended that explains, returning `null` will invalidate the JWT session.